### PR TITLE
Fixes intermittent unit test failures

### PR DESCRIPTION
**Pull request by Aaronius**
## Description
This fixes intermittent unit test failures we were seeing due mostly to uncaught promise rejections.

## Motivation and Context
Passing tests are 👍 .

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: test/unit/specs/utils/dom/awaitSelector.spec.js [hunk 1]
```diff
@@ -31,39 +31,40 @@ describe("DOM::awaitSelector", () => {
...
       .then(nodes => {
-        done();
-        cleanUp(id);
         expect(nodes[0].tagName).toEqual("DIV");
       })
-      .catch(() => {
-        done();
+      .finally(() => {
         cleanUp(id);
-        fail(`${id} should be found`);
```
Comment (Aaronius) [inline test/unit/specs/utils/dom/awaitSelector.spec.js:48]: This one had a couple issues. First, we were calling `done()` here and then `fail`, which didn't really make sense, and the fact that we were calling `done` before cleaning up meant we were potentially attempting to clean up DOM elements after Jasmine moved onto the next test.

File: test/unit/specs/utils/dom/awaitSelector.spec.js [hunk 2]
```diff
@@ -31,39 +31,40 @@ describe("DOM::awaitSelector", () => {
...
-    awaitSelectorAndAssert("abc", window, document, done);
+  it("await via MutationObserver", () => {
+    return awaitSelectorAndAssert("abc", window, document);
   });
 
-  it("await via requestAnimationFrame", done => {
-    const win = { requestAnimationFrame: window.requestAnimationFrame };
+  it("await via requestAnimationFrame", () => {
+    const win = {
+      requestAnimationFrame: window.requestAnimationFrame.bind(window)
```
Comment (Aaronius) [inline test/unit/specs/utils/dom/awaitSelector.spec.js:57]: This was also a reason for failure. `requestAnimationFrame` wasn't working as expected without it being bound to window.

File: test/unit/specs/core/injectExecuteCommand.spec.js [hunk 3]
```diff
@@ -90,25 +90,30 @@ describe("injectExecuteCommand", () => {
...
-    executeCommand("configure");
-    const thenSpy = jasmine.createSpy();
-    const catchSpy = jasmine.createSpy();
-    executeCommand("event")
-      .then(thenSpy)
-      .catch(catchSpy);
+    const configureRejectedSpy = jasmine.createSpy("configureRejectedSpy");
+    executeCommand("configure")
+      .then(fail)
+      .catch(configureRejectedSpy);
```
Comment (Aaronius) [inline test/unit/specs/core/injectExecuteCommand.spec.js:104]: We weren't catching this rejected promise.

File: test/unit/specs/components/Privacy/createComponent.spec.js [hunk 4]
```diff
@@ -154,11 +156,13 @@ describe("privacy:createComponent", () => {
     clearConsentCookie();
     build();
     const setConsentMock = mockSetConsent();
-    component.commands.setConsent.run(CONSENT_IN);
+    const onRejected = jasmine.createSpy("onRejected");
+    component.commands.setConsent.run(CONSENT_IN).catch(onRejected);
```
Comment (Aaronius) [inline test/unit/specs/components/Privacy/createComponent.spec.js:160]: We weren't catching this rejected promise.

**Review by jonsnyder — APPROVED** [review]

**Comment by jonsnyder** [comment]
The remaining 4 test failures are because I'm using "Number.isNaN" in injectExtractEdgeInfo. I will fix this in another PR.

**Review by shammowla — APPROVED** [review]

**Comment by Aaronius** [comment]
Thanks @jonsnyder and @shammowla

**Pull request merged**